pkg/postgres/repository.go

```diff
@@ -425,13 +457,28 @@ WITH station_estimates AS (
   SELECT runs.run_group_identifier,
     estimates.estimated_arrival_time,
     estimates.estimate_moment,
+
+    directions.name,
+    lines.name,
+    stations.name,
+
+    directions.id,
+    lines.id,
+    stations.id,
+
     ROW_NUMBER() OVER (PARTITION BY runs.run_group_identifier
       ORDER BY estimates.estimate_moment DESC) AS rank
   FROM arrivals
   JOIN estimates
     ON estimates.arrival_identifier = arrivals.identifier
   JOIN runs
     ON arrivals.run_identifier = runs.identifier
+  JOIN lines
+    ON lines.id = runs.line_id
+  JOIN directions
+    ON lines.id = runs.direction_id
+  JOIN stations
+    ON lines.id = arrivals.station_id
```

Comment:
Using those fuzzy-matched entity IDs, we fetch the records to return with the API response